nexus/types/src/external_api/params.rs

```diff
@@ -1748,6 +1748,27 @@ pub struct BgpPeer {
 
     /// How often to send keepalive requests (seconds).
     pub keepalive: u32,
+
+    /// Requre that a peer has a specified ASN.
+    pub remote_asn: Option<u32>,
+
+    /// Require messages from a peer have a minimum IP time to live field.
+    pub min_ttl: Option<u8>,
+
+    /// Use the given key for TCP-MD5 authentication with the peer.
+    pub md5_auth_key: Option<String>,
+
+    /// Apply the provided multi-exit discriminator (MED) updates sent to the peer.
+    pub multi_exit_discriminator: Option<u32>,
+
+    /// Include the provided communities in updates sent to the peer.
+    pub communities: Vec<u32>,
+
+    /// Apply a local preference to routes received from this peer.
+    pub local_pref: Option<u32>,
+
+    /// Enforce that the first AS in paths recieved from this peer is the peer's AS.
+    pub enforce_first_as: bool,
```

Comment:
Should these have `serde(default)`?

Reply:
I'm torn on this. I know we did this for the bootstore to make for a smooth upgrade path. However, for the API data structures themselves, I feel like having defaults opens the door to potentially confusing behavior rather than requiring the specification of all parameters in a WYSIWYG model. That way, when this gets hooked up to something like Terraform, when operators look at source-controlled parameter definitions, there are no surprises through defaults.

Reply:
That makes sense -- in that case they're truly optional, so requiring that they be specified as `null` is reasonable.